src/SFA.DAS.AdminService.Web.Tests/Services/Gateway/GatewayOverviewOrchestratorTests.cs

```diff
@@ -0,0 +1,192 @@
+            };
+
+            var returnedRoatpApplicationResponse = new RoatpApplicationResponse
+            {
+                ApplicationId = applicationId,
+                ApplyData = applyData,
+                GatewayReviewStatus = GatewayReviewStatus.InProgress
+            };
+
+            _applyApiClient.Setup(x => x.GetApplication(applicationId)).ReturnsAsync(returnedRoatpApplicationResponse);
+
+            // No Saved Statusses
+            var returnedGatewayPageAnswers = new List<GatewayPageAnswerSummary>();
+
+            _applyApiClient.Setup(x => x.GetGatewayPageAnswers(applicationId)).ReturnsAsync(returnedGatewayPageAnswers);
+
+            var request = new GetApplicationOverviewRequest(applicationId, UserName);
+
+            var response = _orchestrator.GetConfirmOverviewViewModel(request);
+
+            var viewModel = response.Result;
+
+            viewModel.ApplicationId.Should().Be(applicationId);
+            viewModel.Ukprn.Should().Be(ukprn);
+            viewModel.OrganisationName.Should().Be(organisationName);
+            viewModel.GatewayReviewStatus.Should().Be(gatewayReviewStatus);
+            viewModel.ReadyToConfirm.Should().Be(false);
+        }
+
+        [TestCase(GatewayReviewStatus.InProgress, "GatewayReviewStatus", "Error - GatewayReviewStatus")]
+        [TestCase(GatewayReviewStatus.Clarification, "OptionAskClarificationText", "Error - Clarification")]
+        [TestCase(GatewayReviewStatus.Declined, "OptionDeclinedText", "Error - Declined")]
+        [TestCase(GatewayReviewStatus.Approved, "OptionApprovedText", "Error - Approved")]
+        public void ProcessViewModelOnError_process_view_model_correctly(string gatewayReviewStatus, string field, string errorMessage)
+        {
+            var applicationId = Guid.NewGuid();
+
+            var viewModelOnError = new RoatpGatewayApplicationViewModel 
+            {
+                ApplicationId = applicationId,
+                GatewayReviewStatus = gatewayReviewStatus
+            };
+
+            var viewModel = new RoatpGatewayApplicationViewModel
+            {
+                ApplicationId = applicationId,
+                GatewayReviewStatus = gatewayReviewStatus
+            };
+
+            var validationResponse = new ValidationResponse 
+            { 
+                Errors = new List<ValidationErrorDetail> 
+                { 
+                    new ValidationErrorDetail 
+                    { 
+                        Field = field, 
+                        ErrorMessage = errorMessage
+                    } 
+                } 
+            };
+                        
+            _orchestrator.ProcessViewModelOnError(viewModelOnError, viewModel, validationResponse);
+
+            viewModelOnError.ApplicationId.Should().Be(applicationId);
+            viewModelOnError.GatewayReviewStatus.Should().Be(gatewayReviewStatus);
+           
+            if (field.Equals(nameof(viewModelOnError.GatewayReviewStatus)))
```

Comment:
We shouldn't have if statements in tests, they should be separate tests.

Reply:
I have split the ProcessViewModelOnError tests into four individual tests